src/Moq/Extensions.cs

```diff
@@ -72,6 +72,11 @@ public static bool LooksLikeEventDetach(this MethodInfo method)
 			return method.Name.StartsWith("remove_", StringComparison.Ordinal);
 		}
 
+		public static bool ProbablyEventAttachOrDetach(this MethodInfo method)
+		{
+			return method.IsSpecialName && (method.LooksLikeEventAttach() || method.LooksLikeEventDetach());
```

Comment:
Not sure why we need this method at all. Why not simply use the pre-existing methods `LooksLike...`?

You might have noticed that Moq doesn't do the obvious and check `IsSpecialName` when trying to find out if a method is a property or event accessor. There's apparently some history behind this curious lack, which I tried to document: 

https://github.com/moq/moq4/blob/22cce9c3f5a7e6a66f079ad719a7e415636430bb/src/Moq/Interception/InterceptionAspects.cs#L191-L193

I have never looked deeply into the rationale behind 44070a9; perhaps it no longer exists today; I don't know.

* If you know for sure that checking `IsSpecialName` is safe, even in the face of F# and COM code, then I suggest we drop the `ProbablyEventAttachOrDetach` method you're adding, and improve the existing methods instead.

* If OTOH you're not positively sure either, then let's remove your new method and use the existing ones directly.

Reply:
I wasn't aware to the issue about `IsSpecialName`. I don't cae to revert this method and usethe original ones.

Reply:
I think that'd be best for now. I'll do some research this weekend to see if I find out more about why Moq avoids using `IsSpecialName`. I'll post back here about my findings (if any).

Reply:
Here's why Moq does not test for `.IsSpecialName` for events: https://github.com/moq/moq4/pull/712.